### ISSUE TITLE
Original image color was changed

### ISSUE THREAD

**Issue by rubi-du** [ZhengPeng7/BiRefNet#136]
I use the hugging demo to remove image background, but the color changed in image which background was removed.
From these two images, it can be seen that the color of the processed image has become lighter

huggingface space link: https://huggingface.co/spaces/ZhengPeng7/BiRefNet_demo

- original image
<img width="565" alt="image" src="https://github.com/user-attachments/assets/9bfd7b2a-9afb-4b8a-9e17-f79486057cba">

- result

<img width="564" alt="image" src="https://github.com/user-attachments/assets/e9686f2e-b7e5-4dc6-baac-d5b3ca04dce3">

**Comment by rubi-du** [comment]
I found through debugging that there are many floating-point data with very small differences in the mask. Is this the reason for this
<img width="617" alt="image" src="https://github.com/user-attachments/assets/da0c6c1e-fab7-45aa-88dd-edd442b3d637">

**Comment by ZhengPeng7** [comment]
Yeah. If you don't apply hard thresholding, the pixel values of the predicted maps of BiRefNet are actually float ones between 0 and 1. What's more, floats also represent transparency.
